### DISCUSSION TITLE
[conda] A release on anaconda cloud would be very useful

### DISCUSSION THREAD

**Discussion by fg-mindee** [pymupdf/PyMuPDF#1137] [Ideas]
Hey there :wave: 

Thanks for this very useful and high performing PDF manipulation project! I think it could benefit further from extending the range of installation means.

**Is your feature request related to a problem? Please describe.**
I was working on releasing a project which has PyMuPDF as a dependency. The pypi release was rather easy to handle considering you published it over there. However, there is no conda counterpart. I tried to make one on my end, but the complex building dependencies of the project make this difficult.

**Describe the solution you'd like**
Making a release on anaconda cloud so that we can install it with `conda install -c YOUR_CHANNEL pymupdf` (or fitz)

**Describe alternatives you've considered**
Perhaps providing clear instructions about how to build it for conda would be a good first option

**Comment by fg-mindee** [comment]
Hi @JorjMcKie,
Any update on this? :pray:

**Comment by JorjMcKie** [comment]
sorry - I am having a lot of things to do ...

Unfortunately, supporting this is relatively complex - at least in my case:

I did have had a look into this, and it became clear, that I will ever only be able to support platforms, for which I am generating wheels.
This means that a conda install will ultimately lead to a simple wheel installation. There will never be an installation from sources.

That said, I still need to develop scripts for Windows, Linux and Mac OSX, which check the existence of the required wheel and then install it. While I am able to test such scripts for Windows and Linux, I won't be able to do that for Mac.

So bottom line is: **_I need help!_**

**Reply by fg-mindee** [reply]
Hey there :wave:

Thanks again for your project and being the maintainer, I can easily appreciate that knowing all the troubles that come with it :sweat_smile: 
OS-wise, I can only help with Linux and a bit on Windows!

**Comment by JorjMcKie** [comment]
This is what I have so far:
[pymupdf.zip](https://github.com/pymupdf/PyMuPDF/files/7169699/pymupdf.zip)

It contains a Windows batch script `bld.bat` and a Python script `support-check.py`.
* The batch script executes the support checker and either exits with error based on the checker error exit, or invokes `python pip install pymupdf`, because the support checker has determined that a wheel must exists on PyPI.
* similar batch scripts now need to be developed for Linux and Mac
* a suitable `meta.yml` must be developed for conda reflecting the situation.

**Reply by fg-mindee** [reply]
I can take a look at Linux in the coming weeks to try to help :ok_hand:
